ecs-cli/modules/utils/compose/convert_task_definition_test.go

```diff
@@ -136,14 +142,221 @@ task_definition:
 
 	ecsParamsFileName := tmpfile.Name()
 
-	taskDefinition, err := convertToTaskDefWithEcsParamsInTest(t, "name", &config.ServiceConfig{}, "", ecsParamsFileName)
+	taskDefinition, err := convertToTaskDefWithEcsParamsInTest(t, []string{"mysql", "wordpress"}, &config.ServiceConfig{}, "", ecsParamsFileName)
 
 	if assert.NoError(t, err) {
 		assert.Equal(t, "host", aws.StringValue(taskDefinition.NetworkMode), "Expected network mode to match")
 		assert.Equal(t, "arn:aws:iam::123456789012:role/my_role", aws.StringValue(taskDefinition.TaskRoleArn), "Expected task role ARN to match")
+
+		// If no containers are specified as being essential, all
+		// containers are marked "essential"
+		for _, container := range taskDefinition.ContainerDefinitions {
+			assert.True(t, aws.BoolValue(container.Essential), "Expected essential to be true")
+		}
+	}
+}
+
+func TestConvertToTaskDefinitionWithECSParams_Essential_OneContainer(t *testing.T) {
+	ecsParamsString := `version: 1
+task_definition:
+  services:
+    mysql:
+      essential: false`
+
+	content := []byte(ecsParamsString)
+
+	tmpfile, err := ioutil.TempFile("", "ecs-params")
+	assert.NoError(t, err, "Could not create ecs fields tempfile")
+
+	defer os.Remove(tmpfile.Name())
+
+	_, err = tmpfile.Write(content)
+	assert.NoError(t, err, "Could not write data to ecs fields tempfile")
+
+	err = tmpfile.Close()
+	assert.NoError(t, err, "Could not close tempfile")
+
+	ecsParamsFileName := tmpfile.Name()
+
+	taskDefinition, err := convertToTaskDefWithEcsParamsInTest(t, []string{"mysql", "wordpress"}, &config.ServiceConfig{}, "", ecsParamsFileName)
+
+	containerDefs := taskDefinition.ContainerDefinitions
+	mysql := findContainerByName("mysql", containerDefs)
+	wordpress := findContainerByName("wordpress", containerDefs)
+
+	if assert.NoError(t, err) {
+		assert.False(t, aws.BoolValue(mysql.Essential), "Expected essential value to match")
```

Comment:
I'd suggest changing all of the assert messages to be more descriptive, like in this case "Expected essential to be False" or may be "Expected essential on mysql container to be True." If someone ever changes this code and screws up the logic, seeing how the test cases are failing (true vs false being set incorrectly) will make it faster to debug.